PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp

```diff
@@ -1769,7 +1769,7 @@ bool FindReplaceDlg::processFindNext(const TCHAR *txt2find, const FindOption *op
 		}
 
 		if (posFind == -1)

```

Comment:
Could be "IsFailNotFound(posFind)", and "IsFailInvalidRegex(posFind)" below.

Reply:
As indicated here: https://github.com/notepad-plus-plus/notepad-plus-plus/pull/9566#discussion_r594644310, I could imagine to add three constants to  the `scintilla\include\BoostRegexSearch.h`. Something like

	#define ERR_NOT_FOUND (-1)
	#define ERR_WRONG_REGEX (-2)
	#define ERR_SEARCH_FAILED (-3)

But no extra function for this mega-complex issue to detect which of the three error codes has been delivered.

And, as I mentioned before, I believe in generalized error codes and original regex error messages. Honestly, there is a correlation here: Users, who are able to compose the most complicated regex expressions (I ever saw), are certainly able to interpret the original regex error messages and can get more out of the original messages than out of any translation I could provide.